Automattic/pocket-casts-android
# What's new bookmarks dynamic content

## Description

This updates `What's new` dialog's
- Display conditions (https://github.com/Automattic/pocket-casts-android/pull/1460/commits/9f36c52d2c8cd3b7168543d11ebf67bfcbb7dd47)
- Title (d089943b0246d750e3e8288fb0617ec3c6bde897), message (https://github.com/Automattic/pocket-casts-android/pull/1460/commits/87e7b9c3fcd598260c268c40393f71b77a028879), and button title (https://github.com/Automattic/pocket-casts-android/pull/1460/commits/5c5858bfd034f896c10a00c018099864e0263ae5)

based on this grid:

## Screenshots or Screencast 

Release | Free Users | Plus Users | Patron Users
-- | -- | -- | --
Beta early access | - |  ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/facfd9a1-d111-4254-85f6-b759b7afc365)|  ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/1fa48a31-3014-4cbb-bb9b-88813e079c7a)
  |   |   |  
Prod early access |- |  - | ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/871d45cf-d88a-4ba9-949f-6c436cb7f067)
  |   |   |  
Beta full access |![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/7cf2fafa-f677-4585-ad57-0afd51c4822a) | ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/d73f1cd8-ec51-4850-b51e-d9906ec537e5) | ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/526ca65e-d53d-47f4-80fd-87ba64901527)
  |   |   |  
Prod full access | ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/ff1e1d3e-90ca-40b5-ae23-ae4618497aba)  | ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/5aaf524a-4da5-46c4-ac71-57011735d6cc) |  ![image](https://github.com/Automattic/pocket-casts-android/assets/1405144/6a1bb6e9-2e0a-4df0-8b11-b180834f788f)
  |   |   |  


|Free Users, Beta/ Prod full access release with free trial |
|---|
|<img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/4eb59c19-109c-482c-9fc0-8db673a405b3">|


## Testing Instructions

Prerequisites
- Update `patronExclusiveAccessRelease` in `Feature` class
- Install the app

[`Free`] user

1. Increment `Settings.WHATS_NEW_VERSION_CODE`
2. Update version code to [`Beta early access]` release version
3. Re-install the app
4. ✅ Notice that what's new status is as shown in the grid cell for the [Free] user, [Beta early access] release 

Repeat above steps by replacing [`Beta early access]` with [`Prod early access`], [`Beta full access]`, [`Prod full access`]
Repeat above steps by replacing [`Free`] with [`Plus`], [`Patron`] 

Note: 
1. Since the feature will not be enabled before early access release, I haven't tested/ deliberately supported before early access states.
2. Try it now action needs to be updated (p1697115334479459-slack-C055BDMU095)

## Checklist
 
#### I have tested any UI changes...
